libs/gl-client-py/src/credentials.rs

```diff
@@ -0,0 +1,209 @@
+use crate::runtime::exec;
+use crate::scheduler::Scheduler;
+use crate::signer::Signer;
+use crate::TlsConfig;
+use gl_client::credentials;
+use pyo3::exceptions::PyValueError;
+use pyo3::prelude::*;
+
+#[pyclass]
+#[derive(Clone)]
+pub enum CredentialType {
+    Nobody = 0,
+    Device = 1,
+}
+
+#[pyclass]
+#[derive(Clone)]
+pub struct Credentials {
+    pub typ: CredentialType,
+    pub inner: credentials::Credentials,
+    builder: credentials::Builder,
+}
+
+#[pymethods]
+impl Credentials {
+    #[new]
+    pub fn new() -> Result<Self> {
+        Ok(Self {
+            typ: CredentialType::Nobody,
+            inner: credentials::Builder::nobody()?,
+            builder: credentials::Builder::new(),
+        })
+    }
+
+    #[staticmethod]
+    pub fn nobody() -> Result<Self> {
```

Comment:
Is this ever going to change from the implementation of 'new'? maybe we can just call new here so we only have to change the default params in one place

Reply:
fc3e2f6a0524323edc7e4fdbef6e1fe70e63e6de